api/app/src/providers/oauth2.ts

```diff
@@ -140,6 +142,38 @@ export class OAuth2DefaultImpl implements OAuth2 {
     return refreshedToken.access_token;
   }
 
+  async revokeProviderAccess(connectedUser: ConnectedUser) {
+    if (!connectedUser.providerMap) throw new UnauthorizedError();
+    const providerData = connectedUser.providerMap[this.providerName];
+    if (!providerData) throw new UnauthorizedError();
+
+    const client = this.makeClient();
+    const token = JSON.parse(providerData.token);
+    const accessToken = client.createToken(token);
+
+    await accessToken.revoke('access_token');
+
+    updateProviderData({
+      id: connectedUser.id,
+      cxId: connectedUser.cxId,
+      provider: this.providerName,
+      providerItem: undefined
+    })
+  }
+
+  async revokeWithNoOauth(connectedUser: ConnectedUser) {
```

Comment:
A lot of our providers dont have a route to deauthenticate users so i did some research and i think the best we can do is remove from our db and the token will eventually expire

Reply:
Sounds good!

What if we rename this to `revokeLocal` or something, and call it from `revokeProviderAccess` so we only write this logic once? It could return the previous token so `revokeProviderAccess` can use it to do its thing.